cpp/src/lists/contains.cu

```diff
@@ -172,119 +127,66 @@ struct search_list_nested_types_fn {
-                               OutputIterator output_it,
-                               SearchKeyType const& search_keys,
-                               bool search_keys_have_nulls,
-                               duplicate_find_option find_option,
-                               rmm::cuda_stream_view stream)
-{
-  auto const do_search = [=](auto const keys_iter) {
-    thrust::transform(rmm::exec_policy(stream),
-                      input_it,
-                      input_it + num_rows,
-                      keys_iter,
-                      output_it,
-                      search_list_non_nested_types_fn{find_option});
-  };
-
-  if constexpr (search_key_is_scalar) {
-    auto const keys_iter = cudf::detail::make_optional_iterator<Element>(
-      search_keys, nullate::DYNAMIC{search_keys_have_nulls});
-    do_search(keys_iter);
-  } else {
-    auto const keys_cdv_ptr = column_device_view::create(search_keys, stream);
-    auto const keys_iter    = cudf::detail::make_optional_iterator<Element>(
-      *keys_cdv_ptr, nullate::DYNAMIC{search_keys_have_nulls});
-    do_search(keys_iter);
-  }
-}
-
 /**
  * @brief Function to search for index of key element(s) in the corresponding rows of a lists
  * column, specialized for nested types.
  */
-template <bool search_key_is_scalar,
-          typename InputIterator,
-          typename OutputIterator,
-          typename SearchKeyType>
-void index_of_nested_types(InputIterator input_it,
-                           size_type num_rows,
-                           OutputIterator output_it,
-                           column_view const& child,
-                           SearchKeyType const& search_keys,
-                           duplicate_find_option find_option,
-                           rmm::cuda_stream_view stream)
+template <typename InputIterator, typename OutputIterator>
+void index_of(InputIterator input_it,
+              size_type num_rows,
+              OutputIterator output_it,
+              column_view const& child,
+              column_view const& search_keys,
+              duplicate_find_option find_option,
+              rmm::cuda_stream_view stream)
 {
-  // Create a `table_view` from the search key(s).
-  // If the input search key is a (nested type) scalar, a new column is materialized from that
-  // scalar before a `table_view` is generated from it. As such, the new created column will also be
-  // returned to keep the result `table_view` valid.
-  [[maybe_unused]] auto const [keys_tview, unused_column] =
-    [&]() -> std::pair<table_view, std::unique_ptr<column>> {
-    if constexpr (search_key_is_scalar) {
-      auto tmp_column = make_column_from_scalar(search_keys, 1, stream);
-      return {table_view{{tmp_column->view()}}, std::move(tmp_column)};
-    } else {
-      return {table_view{{search_keys}}, nullptr};
-    }
-  }();
-
+  auto const keys_tview  = cudf::table_view{{search_keys}};
   auto const child_tview = table_view{{child}};
```

Comment:
Let's pick whether to use the `cudf::` namespace or not, and be consistent within this function.